libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -0,0 +1,175 @@
+@file:JvmName("NavigationRouteEx")
+
+package com.mapbox.navigation.base.route
+
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.DirectionsWaypoint
+import com.mapbox.api.directions.v5.models.LegStep
+import com.mapbox.api.directions.v5.models.RouteLeg
+import com.mapbox.api.directions.v5.models.RouteOptions
+import java.util.UUID
+
+/**
+ * Wraps a route object used across the Navigation SDK features.
+ *
+ * @param directionsResponse the original response that that returned this route object.
+ * @param routeIndex the index of the route that this wrapper tracks
+ * from the collection of routes returned in the original response.
+ * @param routeOptions options used to generate the [directionsResponse]
+ * @param id unique ID of this route entity used for matching it against additional metadata.
+ * Unique ID doesn't mean that the underlying [directionsRoute] is unique.
+ * Multiple [NavigationRoute]s can represent similar or equal [DirectionsRoute].
+ */
+class NavigationRoute(
+    val directionsResponse: DirectionsResponse,
+    val routeIndex: Int,
```

Comment:
Can you add a builder for this class? We are almost certainly going to want to add something. That will cause a semver if we do right?

For example, what if we want to add the `legIndex`?

Reply:
This is not implemented yet, so I'm definitely open for other ideas.

Reply:
Looks like routes with same `uuid` and `same index` are actually the same, right? We might use it to link NavigationRoute. Or make routeId based on uuid and route index 🤔

Reply:
Generally this is true, but we'd need a valid UUID to be returned from the onboard router as well to make this type of an ID. For all other purposes UUID is theoretically optional because it's only used for refresh, that's why I didn't want to rely on it and make it required (which could be considered as another breaking change).